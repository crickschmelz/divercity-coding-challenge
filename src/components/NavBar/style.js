import styled from "styled-components";

export const Nav = styled.nav`
  background-color: white;
  height: auto;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 10px;
  width: 100%;
`;

export const NavLinkContainer = styled.ul`
  display: flex;
  list-style-type: none;
  margin: 0 3% 0 3%;
  padding: 0;
  overflow: hidden;
  justify-content: flex-end;
  align-items: center;
`;

export const NavLink = styled.button`
  font-size: 16px;
  color: rgb(37, 55, 85);
  margin: 10px 20px;
  text-decoration: none;
  cursor: pointer;
  border: none;
  font-weight: 800;
  background-color: transparent;

  &:after {
    content: "";
    width: 0px;
    height: 1.5px;
    display: block;
    background: hsl(346, 85%, 0%);
    transition: 300ms;
  }

  &:hover:after {
    width: 100%;
  }
`;

export const CreateButton = styled.button`
  width: 200px;
  cursor: pointer;
  align-self: center;
  font-weight: 800;
  font-size: 16px;
  color: rgb(37, 55, 85);
  padding: 10px;
  background-color: transparent;
  border: 1px solid rgb(104, 114, 132);
  margin: 5px 10px;
  border-radius: 4px;
`;
