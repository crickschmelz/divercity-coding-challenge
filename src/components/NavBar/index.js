import React from "react";
import PropTypes from "prop-types";
import { Nav, NavLinkContainer, NavLink, CreateButton } from "./style";

const NavBar = (props) => {
  NavBar.propTypes = {
    clickLink: PropTypes.func
  };

  const { clickLink } = props;

  return (
    <Nav>
      <a href="https://www.divercity.io/">
        <img
          src="https://www.divercity.io/static/media/complete-logo.f047ae08.svg"
          alt="divercity logo"
        ></img>
      </a>
      <NavLinkContainer>
        <li>
          <NavLink onClick={clickLink} value="login">
            Sign in
          </NavLink>
        </li>
        <li>
          <CreateButton onClick={clickLink} value="create-account">
            Create an account
          </CreateButton>
        </li>
      </NavLinkContainer>
    </Nav>
  );
};

export default NavBar;
