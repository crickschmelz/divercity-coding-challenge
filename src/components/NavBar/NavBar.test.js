import React from "react";
import NavBar from "./NavBar";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<NavBar />).toJSON();
  expect(tree).toMatchSnapshot();
});
