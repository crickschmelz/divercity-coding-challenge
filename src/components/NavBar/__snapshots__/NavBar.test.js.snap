// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="sc-AxjAm kVStAK"
>
  <a
    href="https://www.divercity.io/"
  >
    <img
      alt="divercity logo"
      src="https://www.divercity.io/static/media/complete-logo.f047ae08.svg"
    />
  </a>
  <ul
    className="sc-AxirZ AYPIi"
  >
    <li>
      <button
        className="sc-AxiKw evdkuo"
        value="login"
      >
        Sign in
      </button>
    </li>
    <li>
      <button
        className="sc-AxhCb bmKJYB"
        value="create-account"
      >
        Create an account
      </button>
    </li>
  </ul>
</nav>
`;
